product_main_supplierinfo/models/product_product.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    main_seller_id = fields.Many2one(
+        comodel_name="product.supplierinfo",
+        string="Main Vendor",
+        compute="_compute_main_seller_id",
+    )
+
+    @api.depends(
+        "seller_ids.partner_id.active",
+        "seller_ids.sequence",
+        "seller_ids.min_qty",
+        "seller_ids.price",
+        "seller_ids.company_id",
+        "seller_ids.product_id",
+        "seller_ids.date_start",
+        "seller_ids.date_end",
+    )
+    @api.depends_context("company")
+    def _compute_main_seller_id(self):
+        for product in self:
+            sellers = product._get_sellers()
+            product.main_seller_id = fields.first(sellers)
+
+    def _get_sellers(self):
```

Comment:
@adriresu @sebalix My question arrives late but why not using standard method `_get_filtered_sellers()` or _select_seller()` to avoid code duplication ?

Reply:
Good question, had to re-check the original v14 PR to see if I mentioned something there, but nothing!
The only difference that remembers me a thing is that `_select_seller()` is sorting the sellers based on `price` before returning one (which is OK when encoding a PO). Here the price is not a criteria to become the "main vendor" of a product, but sequence and dates matter.

Reply:
Just saw that starting from Odoo v17, `_select_seller` method accepts a new `ordered_by` parameter, which could probably used to force the use of `sequence`, if that works we could drop this method :thinking:

Reply:
It's not that neither, we are sorting on the price at the end of this method too... I don't remember then why I duplicated this code, for sure there is a reason.

Reply:
Find out, so we could not use `_select_seller` in v14 because of this filter:
- https://github.com/odoo/odoo/blame/14.0/addons/product/models/product.py#L637-L638
But starting from v15, which introduced the submethod `_get_filtered_sellers` you were talking about, it should definitly work.

Reply:
Trying this here: https://github.com/OCA/product-attribute/pull/1812
One test is failing, to check.